webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultCsvValidationService.java

```diff
@@ -101,17 +108,23 @@ public DefaultCsvValidationService(final SchoolRepository schoolRepository) {
             final CsvSubject subject = CsvSubject.valueOfCaseInsensitive(record.get(HeaderSubject));
             recordNumber = record.getRecordNumber();
 
-            validateKeyValues(validationFailures, school, group, schoolYear, recordNumber);
-            if (!validationFailures.isEmpty()) continue;
+            validateKeyValues(validationResults, school, group, schoolYear, recordNumber);
+            if (!allOk(validationResults)) continue;
```

Comment:
Hmmm, not a fan of repeatedly iterating over the list...

Reply:
I think I'd rather see a container that has the successes and failures tracked separately.  Thoughts?

Reply:
That's a great idea since I'm already returning separate collections anyways.